tensorboard/webapp/metrics/views/main_view/card_grid_test.ts

```diff
@@ -0,0 +1,188 @@
+import {ScrollingModule} from '@angular/cdk/scrolling';
+import {Component, Input} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+import {PluginType} from '../../data_source';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {CardLazyLoader, CardObserver} from '../card_renderer/card_lazy_loader';
+
+import {CardGridComponent} from './card_grid_component';
+import {CardGridContainer} from './card_grid_container';
+import {CardViewComponent} from '../card_renderer/card_view_component';
+import {CardViewContainer} from '../card_renderer/card_view_container';
+import {CardIdWithMetadata} from '../metrics_view_types';
+import {Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {appStateFromMetricsState} from '../../testing';
+import {State} from '../../../app_state';
+import {selectors as settingsSelectors} from '../../../settings';
+import * as selectors from '../../../selectors';
+
+const scrollElementHeight = 100;
+
+// Utility funciton to allow use of await.
+function timeout(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+@Component({
+  selector: 'testing-component',
+  template: `
+    <div cdkScrollable>
+      <div class="placeholder">placeholder</div>
+      <metrics-card-grid
+        [cardIdsWithMetadata]="cardIdsWithMetadata"
+        [cardObserver]="cardObserver"
+        [groupName]="cardObserver"
+      ></metrics-card-grid>
+      <div class="placeholder">placeholder</div>
+    </div>
+  `,
+  styles: [
+    `
+      div {
+        position: fixed;
+        display: block;
```

Comment:
`div` is already a block element so  you do not need this; here and below.

Reply:
Removed.